src/dotnet/commands/dotnet-add/dotnet-add-p2p/Program.cs

```diff
@@ -13,114 +14,101 @@
+                FullName = LocalizableStrings.AppFullName,
+                Description = LocalizableStrings.AppDescription,
+                HandleRemainingArguments = true,
+                ArgumentSeparatorHelpText = LocalizableStrings.AppHelpText,
+            };
 
-            CommandOption frameworkOption = app.Option(
-                $"-f|--framework <{CommonLocalizableStrings.CmdFramework}>",
-                LocalizableStrings.CmdFrameworkDescription,
-                CommandOptionType.SingleValue);
+            command.HelpOption("-h|--help");
 
-            app.OnExecute(() =>
-            {
-                try
-                {
-                    if (!parentApp.Arguments.Any())
-                    {
-                        throw new GracefulException(CommonLocalizableStrings.RequiredArgumentNotPassed, Constants.ProjectOrSolutionArgumentName);
-                    }
+            command._frameworkOption = command.Option(
+               $"-f|--framework <{CommonLocalizableStrings.CmdFramework}>",
+               LocalizableStrings.CmdFrameworkDescription,
+               CommandOptionType.SingleValue);
 
-                    var projectOrDirectory = parentApp.Arguments.First().Value;
-                    if (string.IsNullOrEmpty(projectOrDirectory))
-                    {
-                        projectOrDirectory = PathUtility.EnsureTrailingSlash(Directory.GetCurrentDirectory());
-                    }
+            return command;
+        }
 
-                    var projects = new ProjectCollection();
-                    var msbuildProj = MsbuildProject.FromFileOrDirectory(projects, projectOrDirectory);
+        public override int Run(string fileOrDirectory)
+        {
+            ProjectCollection projects = new ProjectCollection();
+            MsbuildProject msbuildProj = MsbuildProject.FromFileOrDirectory(projects, fileOrDirectory);
 
-                    if (app.RemainingArguments.Count == 0)
-                    {
-                        throw new GracefulException(CommonLocalizableStrings.SpecifyAtLeastOneReferenceToAdd);
-                    }
+            if (RemainingArguments.Count == 0)
+            {
+                throw new GracefulException(CommonLocalizableStrings.SpecifyAtLeastOneReferenceToAdd);
+            }
 
-                    string frameworkString = frameworkOption.Value();
-                    List<string> references = app.RemainingArguments;
-                    PathUtility.EnsureAllPathsExist(references, CommonLocalizableStrings.ReferenceDoesNotExist);
-                    IEnumerable<MsbuildProject> refs = references.Select((r) => MsbuildProject.FromFile(projects, r));
+            string frameworkString = _frameworkOption.Value();
+            PathUtility.EnsureAllPathsExist(RemainingArguments, CommonLocalizableStrings.ReferenceDoesNotExist);
+            IEnumerable<MsbuildProject> refs = RemainingArguments.Select((r) => MsbuildProject.FromFile(projects, r));
```

Comment:
Since you're doing multiple enumerations of this in the nested loop below and because `MsBuildProject.FromFile` is presumably doing I/O, there's probably a perf benefit to resolving this `ToArray` or `ToList` here.